pandas/io/excel/_base.py

```diff
@@ -774,6 +774,11 @@ def parse(
                         assert isinstance(skiprows, int)
                         row += skiprows
 
+                    if row > len(data) - 1:
+                        raise ValueError(
+                            f"Header index {row} references non-existing rows.",
```

Comment:
Would it be helpful to report the number of rows? E.g.

    f"header index {row} exceeds maximum index {len(data) - 1} of data.",

I also made the h lowercase to match the argument `header` to read_excel.

Reply:
Sounds good, changed